Doc draft PRs for WIP, private forks for CI-bashing (#435)
submariner-io/submariner-website: src/content/development/code-review/_index.en.md

other relevant code has changed, the new job results don't tell us anything new. Since changes are constantly being merged, it's likely
another merge commit and set of jobs will be necessary right before merging anyway.

### Mark work-in-progress PRs as drafts

> To clearly indicate a Pull Request is still under development, not yet ready for review, mark it as a draft. It is not necessary to modify

Reviewer: We reference _PR_ before this so I think we can use it here too.

```suggestion
To clearly indicate a PR is still under development and not yet ready for review, mark it as a draft. It is not necessary to modify
```

> PR summaries or commit messages ("WIP" or "DO NOT MERGE"). Keeping the same PR summary keeps email notifications threaded, and using the

Reviewer: ```suggestion
PR summaries or commit messages to indicate this (eg "WIP" or "DO NOT MERGE"). Keeping the same PR summary keeps email notifications threaded, and using the
```

Author: I had a version like this, but it seemed unnecessarily verbose. Also the "this" is hanging on to a broad idea from the first sentence, which is more difficult to read than necessary IMHO.

Author: I did part of this in the latest.

commit message you plan to merge will allow gitlint to verify it. PRs should typically be marked as drafts if any of the required CI is

> failing.

Reviewer: I'm not clear on the last sentence. If I read it correctly, if a CI job fails that prevents merging we should change it to a draft even if it's ready for review?

Author: I'll try to reword it to be more clear.

### Use private forks for debugging PRs by running CI

If a PR is not expected to pass CI but the author wants to see the results to enable development, use a personal fork to run CI. This avoids
clogging the GitHub Actions job queue of the Submariner-io GitHub Organization. After the same `git push` to your personal fork you'd
typically do for a PR, simply choose your fork as the "base repository" of the PR in GitHub's "Open a pull request" UI. After creating the
PR, CI will trigger as usual but the jobs will count towards your personal queue.

> 

Reviewer: Cool - I didn't know this trick!

Author: Added another line to further clarify.

other relevant code has changed, the new job results don't tell us anything new. Since changes are constantly being merged, it's likely
another merge commit and set of jobs will be necessary right before merging anyway.

### Mark work-in-progress PRs as drafts

To clearly indicate a PR is still under development and not yet ready for review, mark it as a draft. It is not necessary to modify PR
summaries or commit messages (e.g. "WIP", "DO NOT MERGE"). Keeping the same PR summary keeps email notifications threaded, and using the
commit message you plan to merge will allow gitlint to verify it. PRs should typically be marked as drafts if any CI is failing that the
author can fix before asking for code review.

> 

Reviewer 2: ```suggestion
Please do this when opening the PR: instead of clicking on the “Create pull request” button, click on the drop-down arrow next to it, and select “Create draft pull request”. This will avoid notifying code owners; they will be notified when the PR is marked as ready for review.
```

Author: Ah nice, I didn't know this!

### Use private forks for debugging PRs by running CI

If a PR is not expected to pass CI but the author wants to see the results to enable development, use a personal fork to run CI. This avoids
clogging the GitHub Actions job queue of the Submariner-io GitHub Organization. After the same `git push` to your personal fork you'd
typically do for a PR, simply choose your fork as the "base repository" of the PR in GitHub's "Open a pull request" UI. After creating the
PR, CI will trigger as usual but the jobs will count towards your personal queue. You will need to open a new PR against the main repository
once your proposed change is ready for review.

> 

Reviewer 2: ```suggestion
Remember to keep your fork’s main branch up-to-date; this will make your local PR easier to understand.
```
